src/__test__/PatchResolver.spec.js

```diff
@@ -65,6 +61,8 @@ describe('PathResolver', function () {
                     boundary,
                 });
 
+                resolver.handleChunk(`\r\n--${boundary}\r\n`);
+
                 resolver.handleChunk(chunk1);
```

Comment:
Can you add `expect(onResponse).not.toHaveBeenCalled();` above this line to assert the callback is not called after the preamble is sent?